src/app/web3/web3.service.ts

```diff
@@ -42,7 +51,7 @@ export abstract class Web3Service {
 
   /** Reference to the deposit contract */
   get depositContract() {
-    return new this.web3.eth.Contract(DEPOSIT_CONTRACT_ABI as any, DEPOSIT_CONTRACT_ADDRESS);
+    return new this.web3.eth.Contract(DEPOSIT_CONTRACT_ABI as any, this.depositContractAddress);
```

Comment:
I think this is going to fail whenever this.depositContractAddress is not yet defined